src/metrics/prometheus.go

```diff
@@ -163,7 +163,8 @@ func ExportPrometheusMetrics(ctx context.Context, gateways string) {
 }
 
 func pushMetrics(ctx context.Context, gateways []string) {
-	if len(gateways) == 0 {
+	// case of empty string splitted by empty delimiter
+	if len(gateways) == 1 && gateways[0] == "" {
```

Comment:
wouldn't it be safer to check for empty strings when pushing to each gateway? I know we have a vet code at the start of the program but my experience says that such dependencies don't tend to hold for long

Reply:
we choose one gateway randomly below and use it until it failed on push. So if we will have `[""]` and check `gateway != ""`, then will be chosen the same empty gateway in the next iteration. So better to validate input data and then rely on verified correct values. imho

Reply:
Wouldn't it be easier to check for empty string in ExportPrometheusMetrics before calling this function? It's private so nobody would call it otherwise anyway and it makes a better and more intuitive check overall

Reply:
agree. changed